perceval/archive.py

```diff
@@ -0,0 +1,312 @@
+
+    :param: origin: identifier of the repository
+    :param: backend_name: name of the backend
+    :param: backend_version: version of the backend
+    :param: archive_root_path: path where the archives for a given origin,
+            backend name and version are stored
+    """
+
+    def __init__(self, origin, backend_name, backend_version, archive_root_path):
+
+        backend_version = '.'.join(backend_version.split('.')[:2])
+        self.archive_folder_path = os.path.join(archive_root_path, origin, backend_version)
+
+        if not os.path.exists(self.archive_folder_path):
+            os.makedirs(self.archive_folder_path)
+
+        self.backend_name = backend_name
+        self.backend_version = backend_version
+
+    def load(self, fetched_at):
+        """Load an already existing archive
+
+        :param: fetched_at: string representation of a date identifying an archive
+        """
+
+        archive_path = os.path.join(self.archive_folder_path, fetched_at + STORAGE_EXT)
+
+        if not os.path.exists(archive_path):
+            raise ArchiveError(cause="Archive %s not found!" % fetched_at)
+
+        return Archive(archive_path)
+
+    def new(self):
+        """Create a new archive"""
+
+        archive_path = os.path.join(self.archive_folder_path, now_to_str() + STORAGE_EXT)
+        Archive.create(archive_path)
+        return Archive(archive_path)
+
+    def delete(self, target_date):
+        """Delete an archive identified by its date in folder path
+
+        :param: target_date: string representation of a date identifying an archive
+        """
+
+        path = self.archive_folder_path
+        for f in os.listdir(path):
+            if os.path.isfile(os.path.join(path, f)) and f.startswith(target_date) and f.endswith(STORAGE_EXT):
+                os.remove(os.path.join(path, f))
+                break
+
+    def delete_all(self):
+        """Delete all archives in folder path"""
+
+        path = self.archive_folder_path
+        [os.remove(os.path.join(path, f)) for f in os.listdir(path)
+         if os.path.isfile(os.path.join(path, f)) and f.endswith(STORAGE_EXT)]
+
+    def archives(self, from_date=DEFAULT_DATETIME):
+        """List archives in folder path
+
+        :param from_date: date from when the archives can be retrieved
+
+        :return list of archive names
+        """
+
+        path = self.archive_folder_path
+        archives = [f.replace(STORAGE_EXT, "") for f in os.listdir(path)
+                    if os.path.isfile(os.path.join(path, f)) and f.endswith(STORAGE_EXT)]
+
+        return [os.path.join(path, f + STORAGE_EXT) for f in sorted(archives) if str_to_datetime(f) >= datetime_to_utc(from_date)]
+
+
+class Archive:
+    """Basic storage for Perceval.
+
+    This class allows to archive items for a
+    further recovery. Items are stored in a Sqlite db.
+
+    :param archive_path: path where to load/retrieve the archive
+    """
+
+    def __init__(self, archive_path):
```

Comment:
This method should check whether the archive exists and it's a valid database.
It should also load metadata information from the database.